app/controllers/tasks_controller.rb

```diff
@@ -0,0 +1,47 @@
+class TasksController < ApplicationController
+  def index
+    @tasks = Task.all
+  end
+
+  def new
+    @task = Task.new
+  end
+
+  def create
+    task = Task.new(title: params[:task][:title], description: params[:task][:description], date: params[:task][:date])
+    task.save #saves it
+    redirect_to('/tasks')
+  end
+
+  def show
+    @task = Task.find(params[:id])
+  end
+
+  def edit
+    @task = Task.find(params[:id])
+  end
+
+  def update
+    @task = Task.find(params[:id])
+    @task.title = params[:task][:title]
+    @task.description = params[:task][:description]
+    @task.date = params[:task][:date]
+    @task.save
+    redirect_to task_path(@task)
+  end
+
+  def complete
+    @task = Task.find(params[:id])
+    @task.date = Date.today
+    @task.title = @task.title.upcase!
+    @task.save
```

Comment:
Line 36, `@task.title = @task.title.upcase!`, will un-set the task's title when it's marked incomplete! [Turns out Ruby's string `upcase!` method returns `nil` if no changes were made.](https://ruby-doc.org/core-2.4.2/String.html#method-i-upcase-21)

Because you're using the "bang" version, it does change `@task.title`, but then you set `@task.title` to the return value (`nil` if it's already upcase).

The fix here is to use one or the other, but not both:

```ruby
@task.title = @task.title.upcase
# - OR -
@task.title.upcase!
```

FWIW, this was a very surprising fact to me. I would not have designed `upcase!` this way.